### PR TITLE
feat: add enforce linkage checker

### PR DESCRIPTION
**Pull request by athakor**
Towards [#6396](https://github.com/googleapis/google-cloud-java/issues/6396)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -173,6 +178,30 @@
...
+                  </requireMavenVersion>
+                  <requireJavaVersion>
+                    <version>[1.7,)</version>
+                  </requireJavaVersion>
+                  <requireUpperBoundDeps/>
+                </rules>
+              </configuration>
+            </execution>
+          </executions>
+        </plugin>
```
Comment (chingor13) [inline pom.xml]: We already include this check in the shared-config parent project

Comment (athakor) [reply pom.xml]: thanks,for the info.so can we add one more rule over there for linkage checker ?
```
<LinkageCheckerRule implementation="com.google.cloud.tools.dependencies.enforcer.LinkageCheckerRule">
  <reportOnlyReachable>true</reportOnlyReachable>
</LinkageCheckerRule>
```

Comment (elharo) [reply pom.xml]: I don't think you want `<reportOnlyReachable>true</reportOnlyReachable>` here.

**Comment by athakor** [comment]
This enforcer config is already present in the shared-config since at least version 0.3.0 as can be seen [here](https://github.com/googleapis/java-shared-config/blob/v0.3.0/pom.xml#L204-L226)

**Pull request closed without merging**